sklearn/utils/_array_api.py

```diff
@@ -492,6 +492,93 @@ def get_namespace_and_device(
         return xp, False, arrays_device
 
 
+def move_to(*arrays, xp_reference, device_reference):
+    """Move all arrays to `xp_reference` and `device_reference`.
+
+    Each array will be moved to the reference namespace and device if
+    it is not already using it. Otherwise the array is left unchanged.
+
+    `array` may contain `None` entries, these are left unchanged.
+
+    Sparse arrays are supported if the reference namespace is numpy.
+    Otherwise a `TypeError` is raised.
```

Comment:
```suggestion
    Sparse arrays are accepted if the reference namespace is numpy, in which case
    they are returned unchanged. Otherwise a `TypeError` is raised.
```